app/components/contact-page-components/SendMessageForm.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import ErrorMessage from "./ErrorMessage";
+
+const sharedStyles =
+  "mt-4 h-[3.75rem] rounded-[10px] border-[#CCE1FF] bg-white800 px-4 text-white500 dark:border-[#2C3C56] dark:bg-black300 dark:text-white900 lg:rounded-3xl lg:px-6 lg:text-xl";
+
+interface SendMessageFormProps {
+  form: React.RefObject<HTMLFormElement>;
+  sendEmail: (e: React.FormEvent<HTMLFormElement>) => void;
+  removeErrorMessage: (errorMsg: string) => void;
+  errorList: string;
+  formData: {
+    user_name: string;
+    user_email: string;
+    message: string;
+    contact_info: string;
+  };
+  setFormData: React.Dispatch<
+    React.SetStateAction<{
+      user_name: string;
+      user_email: string;
+      message: string;
+      contact_info: string;
+    }>
+  >;
+}
+const SendMessageForm: React.FC<SendMessageFormProps> = ({
+  form,
+  sendEmail,
+  removeErrorMessage,
+  errorList,
+  formData,
+  setFormData,
+}) => {
+  return (
+    <form ref={form} onSubmit={sendEmail} className="flex flex-col lg:w-3/5">
+      <div className="relative flex w-full flex-col">
+        <label className="text-lg font-light text-black300 dark:text-white900 lg:text-xl">
+          {"What's your name?"}
+        </label>
+        <input
+          type="text"
+          name="user_name"
+          className={`input_light dark:input_dark lg:h-20 ${sharedStyles}`}
+          onChange={(e) =>
+            setFormData({ ...formData, user_name: e.target.value })
+          }
+          onClick={() => removeErrorMessage("Name is required")}
+        />
+        {errorList.includes("Name is required") && (
+          <ErrorMessage message="Name is required" />
+        )}
+      </div>
+
+      <div className="relative flex w-full flex-col">
+        <label className="mt-9 text-lg font-light text-black300 dark:text-white900 lg:mt-12 lg:text-xl">
+          {"What's your email?"}
+        </label>
+        <input
+          type="email"
+          name="user_email"
+          className={`input_light dark:input_dark lg:h-20 ${sharedStyles}`}
+          onChange={(e) =>
+            setFormData({ ...formData, user_email: e.target.value })
+          }
+          onClick={() => removeErrorMessage("Invalid email format")}
+        />
```

Comment:
This works for me. No worries... You could, however, provide a helper function that takes in the fieldName, errorMsg, and value... 

```
const handleInputChange = (fieldName: string, errorMsg: string, value: string) => {
  setFormData({ ...formData, [fieldName]: value });
  removeErrorMessage(errorMsg);
};
```

Reply:
I have this updated on my latest commit. Now it's dynamic and reusable